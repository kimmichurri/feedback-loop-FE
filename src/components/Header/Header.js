import React, { Component } from 'react';
import Logo from '../../assets/turinglogo.png'
import { Link } from "react-router-dom";

export default class Header extends Component {
  constructor(props) {
    super(props);
    this.state = {
    }
  }

  handleClick = (e) => {
    localStorage.clear();
    this.forceUpdate();
  }

  render() {
    return(
      <div className='header-container'>
        <section className='nav-left'>
          <a href='https://turing.io/'><img className='turing-logo' src={Logo} alt="logo"/></a>
          <Link to='/' className='feedback-loop-title'>
            <h2 className='feedback-loop-title-text'>Feedback Loop</h2>
          </Link>
        </section>
        { localStorage.getItem('currentUser') &&
          <ul className='nav-right'>
            <li>
              <a href='' className='nav-link'>Dashboard</a>
            </li>
            <li>
              <a href='' className='nav-link'
                         onClick={this.handleClick}>Log Out</a>
            </li>
          </ul>
         }
      </div>
    )
  }
}
