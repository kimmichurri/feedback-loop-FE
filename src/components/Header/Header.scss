.header-container {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  margin: 1% 2% 1% 2%;
  align-items: center;
  height: 12vh;
  min-height: 5.3125rem;
}

.nav-left {
  display: flex;
  align-items: center;
  min-width: 530px;
}

.nav-right {
  display: flex;
  flex-direction: row;
  list-style-type: none;
  margin: 0;
  padding: 0;
  align-self: flex-end;
  width: 40%;
  justify-content: flex-end;
  margin-bottom: 2%;
  margin-right:2%;
  li {
    width: 100px;
  }
}

.nav-right a {
  color: #5f5f5f;
  text-decoration: none;
  -webkit-transition: border-color .333s;
  transition: border-color .333s;
  font-weight: bold;
  margin-left: 10%;
  text-decoration: none;
  width: 40%;
  letter-spacing: -1px;
  text-transform: uppercase;
  &:hover {
    border-bottom: 3px solid $main-blue;
  }
}

.turing-logo {
  height: 5.3125rem;
  width: 5.3125rem;
  padding-right: 15%;
  padding-left: 15%;
}

.feedback-loop-title-text {
  font-size: 60px;
  color: $main-grey;
  font-weight: bolder;
  justify-content: left;
  margin: 0px;
  text-decoration: none;
  letter-spacing: -3px;
}

.feedback-loop-title {
  text-decoration: none;
  align-items: center;
  margin-left: 0%;
}
