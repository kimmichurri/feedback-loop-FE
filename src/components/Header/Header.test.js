import React from 'react'
import Header from './Header'
import { shallow } from 'enzyme'

describe('Header', () => {
  let wrapper
  it('should match the snapshot', () => {
    wrapper = shallow(<Header />)
    expect(wrapper).toMatchSnapshot()
  })
})