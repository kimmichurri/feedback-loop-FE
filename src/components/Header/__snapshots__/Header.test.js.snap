// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match the snapshot 1`] = `
<div
  className="header-container"
>
  <section
    className="nav-left"
  >
    <a
      href="https://turing.io/"
    >
      <img
        alt="logo"
        className="turing-logo"
        src="turinglogo.png"
      />
    </a>
    <Link
      className="feedback-loop-title"
      to="/"
    >
      <h2
        className="feedback-loop-title-text"
      >
        Turing Feedback
      </h2>
    </Link>
  </section>
</div>
`;
